### PR TITLE
PR: Sync Outline Explorer at startup and preserve the file order in the tabbar

### LINKED ISSUE #7338: Outline explorer not synchronized at startup
A small GUI bug.

When starting spyder with several files open, and the "structure panel" switched on. The GUI ends up, just after booting, with first file tab and apparently last parsed structure (from last tab), hence leading to a non synchronization between the 2. Switching twice file tabs resynchronizes everything hopefully.

(current mainstream)

### PR DESCRIPTION
**Pull request by jnsebgosselin**
### Issue(s) Resolved

Fixes #7338

### Pull Request Checklist

* [x] Added at least one unit test covering the changes, if at all possible
* [x] Described the changes and the motivation for them below
* [x] Included a screenshot, if this PR makes any visible changes to the UI

### Before :

image

### After :

image

### Developer Certificate of Origin Affirmation

By submitting this Pull Request or typing my name below, I affirm the
[Developer Certificate of Origin](https://developercertificate.org/)
with respect to both the content of the contribution itself and this post,
and understand I am releasing it under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Jean-Sébastien Gosselin

### CODE REVIEW

**Comment by ccordoba12** [comment]
@jnsebgosselin, please fix the failure in Travis, it was introduced by a change done in this PR.

File: spyder/plugins/editor/plugin.py [hunk 1]
```diff
@@ -2959,39 +2973,6 @@ def setup_open_files(self):
             self.__load_temp_file()
         self.set_create_new_file_if_empty(True)
 
-    def reorder_filenames(self, filenames):
```
Comment (ccordoba12) [inline spyder/plugins/editor/plugin.py]: This method seems to cover more things than what your current code does (e.g. it seems to handle splitted editor stacks). Please explain why you're removing it.

File: spyder/plugins/editor/tests/test_plugin.py [hunk 2]
```diff
@@ -63,36 +80,40 @@ def test_basic_initialization(setup_editor):
...
-        (['/file1.py', '/file2.py', '/file3.py', '/file4.py'],
-         ['/file2.py', '/file1.py', '/file3.py', '/file4.py'],
-         [(False, '/file2.py', [2, 4, 8, 12])],
-         [(False, '/file2.py', [4, 2, 8, 12])]),
-        (['/file1.py', '/file2.py', '/file3.py', '/file4.py'],
-         ['/file1.py', '/file2.py', '/file3.py', '/file4.py'],
-         None,
-         None, )
-    ])
-def test_reorder_filenames(setup_editor, expected_filenames, filenames,
```
Comment (ccordoba12) [inline spyder/plugins/editor/tests/test_plugin.py]: This test seems more comprehensive than the test you're replacing it with. Please explain why you're replacing it.

File: spyder/plugins/editor/plugin.py (hunk 1 above)
Comment (jnsebgosselin) [reply spyder/plugins/editor/plugin.py]: The **only** thing this method does, is to load the `splitsettings`, take the filename that had focus last in the first `EditorStack` (`index = 0`), move it to position 0, and save the modified `splitsettings`. 

The settings for the additional splitted editor stacks (`index > 0`) are not modified by this.

This PR proposes to preserve the file order that was saved when Spyder was last closed and to stop reordering files during the setup. Therefore, with the proposed changes, this method is not used anywhere else in the code. I can keep it however if you prefer in case it is needed for something else at some point.

Comment (ccordoba12) [reply spyder/plugins/editor/plugin.py]: But does focus jumps from one file to another while the Editor is opening them? That's my main concern here.

File: spyder/plugins/editor/tests/test_plugin.py (hunk 2 above)
Comment (jnsebgosselin) [reply spyder/plugins/editor/tests/test_plugin.py]: I'm not actually replacing it. It is more like I'm removing `test_reorder_filenames` that was checking only the output of the `reorder_filenames` method and I'm creating a new test that is checking the results of `setup_open_files` instead. So in fact, I think the new test is more comprehensive because it actually checks the resulting order of the files in the `Editorstack` and also check that the right file is focused.

The only thing I do not cover compared to the old test, is when `splitsettings` is `None`, because unless I'm mistaken and I can be, this case is not supposed to happen at this point in the setup of Spyder, so there is no point in testing it.

I simplified the `@pytest.mark.parametrize` because some of the arguments were not needed and there was also duplication of information. Maybe that is why this test looks less comprehensive?

File: spyder/plugins/editor/plugin.py (hunk 1 above)
Comment (jnsebgosselin) [reply spyder/plugins/editor/plugin.py]: No it does not. I'll post an animation to show you what it is doing, one sec.

**Comment by jnsebgosselin** [comment]
setup_files

File: spyder/plugins/editor/plugin.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/plugins/editor/plugin.py]: Ok, that's good. Preserving file order and loading the last file with focus first is ice, UX-wise.

@CAM-Gerlach, what do you think?

Comment (jnsebgosselin) [reply spyder/plugins/editor/plugin.py]: The Outline Explorer is jumping around though, but it was doing that before.At least with this PR, it ends up showing the tree of the right file at the end of the setup. I can fix this in another PR if you think it should.

Comment (CAM-Gerlach) [reply spyder/plugins/editor/plugin.py]: Yeah, that's fine and does look a little nicer at startup as well, although since its only transient its not the biggest concern for me personally.

Of course, the outline explorer is still ordered alphabetically, rather than the order of the tabs in the Editor. Presumably out of scope for now, but if/when you make that change, if would be nice to have the option to either alphabetize it or make it follow the order of the editor, since both could potentially be useful depending on one's workflow.

Also, to note, the outline explorer only show Python files, so it isn't a true file switcher for more general cases. Perhaps, as a longer-term idea, we could have an option to show them (just the files without the outlines, of course).

Comment (jnsebgosselin) [reply spyder/plugins/editor/plugin.py]: > Yeah, that's fine and does look a little nicer at startup as well, although since its only transient its not the biggest concern for me personally.

The most important part of this change is not the transient effect, but the fact that we do not need to reorder the tabs during setup.

File: spyder/plugins/editor/tests/test_plugin.py (hunk 2 above)
Comment (ccordoba12) [reply spyder/plugins/editor/tests/test_plugin.py]: Ok, thanks for the explanation.

Comment (jnsebgosselin) [reply spyder/plugins/editor/tests/test_plugin.py]: No problem. Thanks for the review, it is always appreciated.
